src/jit/compiler.cpp

```diff
@@ -4594,6 +4594,8 @@ void Compiler::compCompile(void** methodCodePtr, ULONG* methodCodeSize, JitFlags
     // this point is responsible for calling fgSetStmtSeq() to keep the
     // nodes properly linked.
     // This can create GC poll calls, and create new BasicBlocks (without updating dominators/reachability).
+
+    fgUpdateSafepointFlags();
```

Comment:
You should add a comment explaining why you placed this call here:

// This needs to be done before we call fgSetBlockOrder because that method can call  fgCreateGCPolls()
//  which uses the BBF_GC_SAFE_POINT flag to determine if we need to insert GC Poll helper calls.